src/Paprika/Store/PageManagers/MemoryMappedPageManager.cs

```diff
@@ -61,11 +61,9 @@ public unsafe MemoryMappedPageManager(long size, byte historyDepth, string dir,
             _file = new FileStream(Path, FileMode.Open, FileAccess.ReadWrite, FileShare.None, 4096,
                 PaprikaFileOptions);
         }
+        _mapped = MemoryMappedFile.CreateFromFile(_file, null, 0, MemoryMappedFileAccess.ReadWrite, HandleInheritability.None, true);
```

Comment:
AFAIK, the mapping will get size from the `_file`, but if the file is just created, it won't work as expected as one will need to set the length. I don't understand how this should work.

Reply:
what if we remove this line - 

_file.setLength(size);

while creating a new file. We can set it to some safe initial size.

Reply:
I don't know how to answer. We may do all the things as long as the file is properly mapped and written and flushed 😅

Reply:
Would you recommend - 

1. initialize the _file with an initial size of say, 4 MB. 
2. modifying the write and flush operations in a way it first checks the size of the _file and only if it's large enough to store incoming data, proceed as it is. Otherwise, can double the file size until it can accommodate incoming data.

Reply:
> Would you recommend -
> 
> 1. initialize the _file with an initial size of say, 4 MB.
> 2. modifying the write and flush operations in a way it first checks the size of the _file and only if it's large enough to store incoming data, proceed as it is. Otherwise, can double the file size until it can accommodate incoming data.

@Scooletz  What do you think about this approach?